src/Hooks/RouteMiddleware.php

```diff
@@ -31,6 +31,17 @@ public function handle(Request $request, Closure $next): mixed
             $this->nightwatch->report($e, handled: true);
         }
 
-        return $next($request);
+        $response = $next($request);
+
+        // If an exception occurs in the action phase, the usual
+        // ResponsePrepared event is not fired. This fallback
+        // ensures that we go to the AfterMiddleware stage.
+        try {
+            $this->nightwatch->stage(ExecutionStage::AfterMiddleware);
```

Comment:
When an unhandled exception occurs, is the `PreparingResponse` event also not fired, and does it skip the `Render` stage as well?